>From 17b9608868ecce851f416518a944314d6ab86973 Mon Sep 17 00:00:00 2001
From: Petr Vorel <petr.vorel@gmail.com>
Date: Wed, 12 Mar 2014 21:27:46 +0100
Subject: [PATCH] fix locale bug when locale file is localized, but encoding is
 wrong

Test with
export LC_ALL=C LANGUAGE=cs_CZ.UTF-8
Then texts are from po/cs.po, but charset in header is "ANSI_X3.4-1968"
---
 lib/apt-listbugs/logic.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/apt-listbugs/logic.rb b/lib/apt-listbugs/logic.rb
index d73051c..573b513 100644
--- a/lib/apt-listbugs/logic.rb
+++ b/lib/apt-listbugs/logic.rb
@@ -101,7 +101,7 @@ class AppConfig
 
     @arrow = "->"
     @xarrow = "-&gt;"
-    if Locale.charset == "UTF-8"
+    if Locale.current.charset == "UTF-8"
       @arrow = "→"
       @xarrow = "→"
     end
@@ -649,7 +649,7 @@ class Viewer
     def display_bugs_as_html(bugs, cur_pkgs, new_pkgs)
       tmp = Tempfile.new(["apt-listbugs", ".html"])
       tmp.chmod(0644)
-      tmp.puts "<?xml version=\"1.0\" encoding=\"#{Locale.charset}\"?>"
+      tmp.puts "<?xml version=\"1.0\" encoding=\"#{Locale.current.charset}\"?>"
       tmp.puts "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\""
       tmp.puts "    \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">"
       tmp.puts "\n<html xmlns=\"http://www.w3.org/1999/xhtml\">"
-- 
1.9.0.279.gdc9e3eb

